### PR TITLE
`rgh-netiquette` - Support new views

### LINKED ISSUE #8012: `rgh-*` features broken on new beta views
I listed all the private features:

- [x] `rgh-linkify-features`
	- Maybe related to https://github.com/refined-github/refined-github/issues/7856
- [ ] `rgh-netiquette`
	- ~~Related to and maybe fixed by https://github.com/refined-github/refined-github/pull/8010~~
- [x] `rgh-dim-commits`
	- Related to https://github.com/refined-github/refined-github/issues/7808
- [x] `rgh-feature-descriptions`
- [x] `rgh-options-link`
- [x] `rgh-improve-new-issue-form`
- [x] `rgh-linkify-yolo`
- [x] `rgh-pr-template`
- [x] `rgh-token-user`

Those unchecked are broken or not yet checked on the new views.

### PR DESCRIPTION
**Pull request by kovsu**
- closes #8012

## Test URLs

https://github.com/download-directory/download-directory.github.io/issues/74

## Screenshot

image

### CODE REVIEW

**Comment by kovsu** [comment]
image

I have tried bleow code and add some class for `TimeLineItem`, but it doesn't work. 

```tsx
export default function TimelineItem({
	className,
}: {
	className?: string;
}): JSX.Element {}
```

And it thorw an error: `TypeError: Cannot read properties of undefined`

**Comment by kovsu** [comment]
I also check the `dom-chef` lib, I don't find anything useful for me.

**Comment by fregante** [comment]
Unlike React components, any attributes you pass to dom-chef JSX will automatically be added. For classes, they will be appended. So `<TimelineItem className='something'>` will just work.

However you most likely just want to change the `TimelineItem` itself since we do want it to stay compatible across all the features that use it (e.g. `closing-remarks`)

Wherever possible, when you make changes to components or classes, search the codebase for the keyword and you might be able to fix multiple features at once. e.g. like you did for `color-fg-muted` in https://github.com/refined-github/refined-github/pull/7313

**Comment by kovsu** [comment]
> Unlike React components, any attributes you pass to dom-chef JSX will automatically be added. For classes, they will be appended. So <TimelineItem className='something'> will just work.

As you can see I add `ml-md-0` to remove the margin, but it be overrode by `ml-md-6`. This is the reason why I want to have a customize classname here.

image

**Comment by kovsu** [comment]
> However you most likely just want to change the TimelineItem itself since we do want it to stay compatible across all the features that use it (e.g. [closing-remarks](https://github.com/refined-github/refined-github/blob/main/source/features/closing-remarks.tsx))
>
>Wherever possible, when you make changes to components or classes, search the codebase for the keyword and you might be able to fix multiple features at once. e.g. like you did for color-fg-muted in https://github.com/refined-github/refined-github/pull/7313

image

What's your opinion fro this way? It's a liitle bit wired for me.

**Comment by fregante** [comment]
> As you can see I add `ml-md-0` to remove the margin, but it be overrode by `ml-md-6`

Since the component needs to work in both layouts, then the right class needs to be applied depending on context. `TimelineItem` is very simple so you can just duplicate it and export a new version just for the new timeline, e.g.

```tsx
export function TimelineItemOld(): JSX.Element {
	// Classes copied from #issuecomment-new + mt-3 added
	return <div className="ml-0 pl-0 ml-md-6 pl-md-3 mt-3" />;
}

!! adjust/remove all classes as needed
export function TimelineItem(): JSX.Element {
	return <div className="ml-0 pl-0 ml-md-0 pl-md-3 mt-3" />;
}
```

And then in the features you can use:

```jsx
const Wrapper = yourLogicToDetectOldView ? TimelineItemOld : TimelineItem;

const banner = (
	<Wrapper>{createBanner({etc})}</Wrapper>
)
```

You can alternatively use the `createBanner` style instead of JSX, like:

```js
const banner = createTimelineItem(
	yourLogicToDetectOldView ? 'old' : 'react',
	createBanner({etc})
)
```

Up to you

**Comment by kovsu** [comment]
image

image

File: source/github-helpers/timeline-item.tsx [hunk 1]
```diff
@@ -1,6 +1,11 @@
 import React from 'dom-chef';
 
-export default function TimelineItem(): JSX.Element {
+export function TimelineItemOld(): JSX.Element {
 	// Classes copied from #issuecomment-new + mt-3 added
 	return <div className="ml-0 pl-0 ml-md-6 pl-md-3 mt-3" />;
 }
+
+// #8141
```
Comment (fregante) [inline source/github-helpers/timeline-item.tsx:8]: I've been preferring full URLs unless it's just minor information. Here you left no other comment so this makes it easier to get more context.

```suggestion
// https://github.com/refined-github/refined-github/pull/8141
```

File: source/github-helpers/timeline-item.tsx [hunk 2]
```diff
@@ -1,6 +1,11 @@
...
+export function TimelineItem(): JSX.Element {
+	return <div className="ml-0 pl-0 pl-md-3 mt-3" />;
```
Comment (fregante) [inline source/github-helpers/timeline-item.tsx:10]: Let's make sure it works correctly on all window widths. These classes were for the old version

**Review by fregante — APPROVED** [review]
LGTM other than the 2 comments, but don't merge it yet. I'll likely release a new version tomorrow with just #8143

File: source/github-helpers/timeline-item.tsx [hunk 3]
```diff
@@ -1,6 +1,11 @@
...
 
-export default function TimelineItem(): JSX.Element {
+export function TimelineItemOld(): JSX.Element {
 	// Classes copied from #issuecomment-new + mt-3 added
 	return <div className="ml-0 pl-0 ml-md-6 pl-md-3 mt-3" />;
 }
+
+// https://github.com/refined-github/refined-github/pull/8141
+export function TimelineItem(): JSX.Element {
+	return <div className="my-2" />;
```
Comment (kovsu) [inline source/github-helpers/timeline-item.tsx:10]: It seems it only need `my-2` in new view. It works on all widths.

https://github.com/user-attachments/assets/976d2db5-95cb-489b-8901-a875206e9c7a

**Comment by fregante** [comment]
Wait, to me it looks like it's in the wrong container. The banner should not go under the sidebar for example, it should be part of the main thread column

**Comment by kovsu** [comment]
> Wait, to me it looks like it's in the wrong container. The banner should not go under the sidebar for example, it should be part of the main thread column

No, it just on the front of comment box.

**Review by fregante — APPROVED** [review]
Might be good to follow this with `closing-remarks`

**Pull request merged**